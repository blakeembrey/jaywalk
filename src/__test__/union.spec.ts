import { validationFunction } from '../'

describe('Union', () => {
  it('should validate union', () => {
    const validate = validationFunction({
      '@type': 'Union',
      type: [
        {
          '@type': 'String'
        },
        {
          '@type': 'Number'
        }
      ]
    })

    const input = 'testing'

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should reject invalid unions', () => {
    const validate = validationFunction({
      '@type': 'Union',
      type: [
        {
          '@type': 'String'
        },
        {
          '@type': 'Number'
        }
      ]
    })

    expect.assertions(2)

    return validate(true).catch((err) => {
      expect(err).toBeInstanceOf(Error)
      expect(err.errors).toMatchSnapshot()
    })
  })
})
