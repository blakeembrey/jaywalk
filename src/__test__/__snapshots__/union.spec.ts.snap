// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Union should reject invalid unions 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "not a string",
    "schemaPath": Array [
      "type",
      0,
      "@type",
    ],
  },
  Object {
    "dataPath": Array [],
    "message": "not a number",
    "schemaPath": Array [
      "type",
      1,
      "@type",
    ],
  },
]
`;
